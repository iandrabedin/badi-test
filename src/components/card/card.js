import React from "react";
import "./card.scss";

export const Card = props => {
  const { url, image, title, ingredients } = props;

  return (
    <div className="col-sm-12 col-md-6">
      <div className="card">
        <div className="image">
          <a href={url} target="_blank" rel="noopener noreferrer">
            <img src={image} alt={title} />
          </a>
        </div>
        <div className="info">
          <h3 className="title">{title}</h3>
          <p className="ingredients">{ingredients}</p>
        </div>
      </div>
    </div>
  );
};
