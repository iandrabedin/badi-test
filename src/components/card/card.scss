@import "../../styles/variables.scss";

.card {
  box-shadow: 0 2px 8px rgba(0, 0, 0, 0.1);
  margin: 12px 0;

  .image {
    display: flex;
    position: relative;
    overflow: hidden;

    img {
      cursor: pointer;
      height: 140px;
    }

    .tag {
      color: $color-white;
      background-color: rgba($color-primary, 0.8);
      text-align: center;
      width: 280px;
      height: 40px;
      line-height: 40px;
      top: 18px;
      right: -13%;
      position: relative;
      -webkit-transform: rotate(45deg);
      -ms-transform: rotate(45deg);
      transform: rotate(45deg);
    }
  }

  .info {
    padding: 8px 16px;

    .title {
      color: $color-primary;
      font-weight: $font-weight-bold;
      font-size: $font-size-x-large;
      margin: 0;
    }

    .ingredients {
      color: $color-grey-dark;
      font-weight: $font-weight-semibold;
      font-size: $font-size-small;
      text-transform: capitalize;
      margin: 0;
      padding: 4px 0;
    }
  }
}

// Tablet - Desktop
@media (min-width: 768px) {
  .card {
    margin: 0 24px 24px 0;

    .image {
      .tag {
        top: 40px;
        right: -25%;
      }
    }
  }
}
