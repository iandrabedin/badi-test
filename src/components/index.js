export { CardsList } from "./cardsList";
export { Card } from "./card";
export { Form } from "./form";
