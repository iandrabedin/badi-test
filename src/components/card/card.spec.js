import React from "react";
import { render, cleanup } from "@testing-library/react";
import { Card } from "./card";
import { handleLactose, htmlDecode } from "../../components";

afterEach(cleanup);

const props = {
  id: 1,
  url: "url",
  image: "image",
  title: "title",
  ingredients: "eggs",
  lactose: false
};

describe("Card", () => {
  it("should render as expected", () => {
    const { asFragment } = render(<Card {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should not show tag if there is not lactose", () => {
    const lactoseProps = {
      ...props,
      lactose: handleLactose(props.ingredients)
    };
    const { queryByText } = render(<Card {...lactoseProps} />);
    expect(queryByText("Has Lactose")).not.toBeInTheDocument();
  });

  it("should show tag if there is lactose", () => {
    const lactoseProps = { ...props, lactose: handleLactose("milk") };
    const { queryByText } = render(<Card {...lactoseProps} />);
    expect(queryByText("Has Lactose")).toBeInTheDocument();
  });

  it("should decode html entitity to correct title", () => {
    const entitityProps = { ...props, title: htmlDecode("Souffl&#233;") };
    const { queryByText } = render(<Card {...entitityProps} />);
    expect(queryByText("Souffl&#233;")).not.toBeInTheDocument();
    expect(queryByText("SoufflÃ©")).toBeInTheDocument();
  });

  it("should show title and ingredients", () => {
    const { getByText } = render(<Card {...props} />);
    expect(getByText("title")).toBeInTheDocument();
    expect(getByText("eggs")).toBeInTheDocument();
  });
});
