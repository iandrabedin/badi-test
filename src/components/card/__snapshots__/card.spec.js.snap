// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render as expected 1`] = `
<DocumentFragment>
  <div
    class="col-sm-12 col-md-6"
  >
    <div
      class="card"
    >
      <div
        class="card__image"
      >
        <a
          href="url"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="title"
            src="image"
          />
        </a>
      </div>
      <div
        class="card__info-wrapper"
      >
        <h3
          class="card__title"
        >
          title
        </h3>
        <p
          class="card__description"
        >
          eggs
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
